src/ploigos_step_runner/step_implementers/shared/rekor_generic.py

```diff
@@ -0,0 +1,285 @@
+"""`StepImplementer` for the `automated-governance` step using Rekor.
+
+Step Configuration
+------------------
+Step configuration expected as input to this step.
+Could come from:
+
+  * static configuration
+  * runtime configuration
+  * previous step results
+
+Configuration Key       | Required? | Default | Description
+-------------------     |-----------|---------|-----------
+`rekor-server-url`      | Yes       |         | URL for Rekor server to upload artifact(s) to.
+`signer-pgp-private-key`| Yes       |         | PGP Private Key used to sign the image
+
+Result Artifacts
+----------------
+Results artifacts output by this step.
+
+Result Artifact Key | Description
+--------------------|------------
+`rekor-uuid`        | UUID of Rekor entry created by artifact upload.
+`rekor-entry`       | DEPRECIATED. Will likely be removed in next release.
+"""
+
+from base64 import b64encode
+import json
+import sys
+from io import StringIO
+import sh
+from ploigos_step_runner import StepImplementer, StepResult
+from ploigos_step_runner.utils.file import base64_encode, get_file_hash,\
+    download_source_to_destination
+from ploigos_step_runner.utils.io import \
+    create_sh_redirect_to_multiple_streams_fn_callback
+from ploigos_step_runner.utils.pgp import detach_sign_with_pgp_key
+from ploigos_step_runner.utils.pgp import import_pgp_key
+from ploigos_step_runner.utils.pgp import export_pgp_public_key
+
+
+DEFAULT_CONFIG = {}
+
+REQUIRED_CONFIG_OR_PREVIOUS_STEP_RESULT_ARTIFACT_KEYS = [
+    'rekor-server-url',
+    'signer-pgp-private-key'
+]
+class RekorGeneric(StepImplementer):  # pylint: disable=too-few-public-methods
```

Comment:
@jthompkins can you rename this to `RekorSignGeneric`. Realized that in our next phase of work we will need a `RekorVerifySignatureGeneric` which verifies a given object is signed by a given public key. So might as well avoide name conflcits now.

Reply:
Changed.